client/server/pages/index.js

```diff
@@ -140,7 +148,7 @@ function getDefaultContext( request, response, entrypoint = 'entry-main' ) {
 
 	const oauthClientId = request.query.oauth2_client_id || request.query.client_id;
 	const isWCComConnect =
-		( 'login' === request.context.sectionName || 'signup' === request.context.sectionName ) &&
+		( 'login' === sectionName || 'signup' === sectionName ) &&
```

Comment:
Drive-by-fix: context.sectionName is undefined everywhere in this function, so there's no way this was working.